app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -49,6 +49,13 @@ export default class SchoolDisciplineDashboard extends React.Component {
     this.memoize = memoizer();
   }
 
+  //Remove scatter plot option when there are too many incidents to show patterns
+  componentDidMount() {
+    if (this.getIncidentsFromStudents(this.props.dashboardStudents).length > 500) {
+      this.setState({selectedChart: 'incident_location', scatterPlotAvailable: false});
+    }
+  }
+
```

Comment:
Can you say more about why this is here, rather than computed in `render`?  I don't follow why it's using the lifecycle hook.

Reply:
It's here because the state has to change (from default scatter to some alternative), which can't be done in render. I could probably set the initial state conditionally in the constructor, but it seemed cleaner to just use getIncidentsFromStudents in the mounted component. It shouldn't be much of a performance issue since we cache the result of that function with memoizer.

Reply:
ah, thanks!  Got it, it's because this is set in `initialState` in the constructor, that's super helpful.  Yeah, the better way to do this is describe this explicitly in the constructor, and then let `initialState` decide what `selectedChart` should be based on those props.  Here that would involve factoring out the incident and filtering functions to be plain functions (rather than relying on `this.state` and `this.props`) and that seems a larger change than this is worth.  Using the lifecycle hook implies that it's because of something related to the DOM or outside of React, and then leads to double-rendering on load.

If you tested this and it works without a visible jump, let me know and let's 🚢 as is.  And thanks for explaining and helping me learn! 👍

Reply:
Great! I have tested this locally with schools above and below the 500 student threshold, and in the former case the scatter plot is cleanly removed. I would like to double check in production just to be sure the threshold is appropriate, but that can probably come from feedback from users.

Reply:
@edavidsonsawyer awesome, thanks! 🚢 ing now and I'll verify as well when it's out.